src/analysis/complex/conformal.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import analysis.complex.isometry
+import analysis.calculus.conformal
+
+/-!
+# Complex Conformal Maps
+
+We prove the sufficient and necessary conditions for a complex function to be conformal.
+
+## Main results
+
+* `is_complex_or_conj_complex_linear_iff_is_conformal_map`: the main theorem for linear maps.
+* `conformal_at_iff_holomorphic_or_antiholomorph`: the main theorem for complex functions.
+
+## Warning
+
+Antiholomorphic functions such as the complex conjugate are considered as conformal functions in
+this file.
+-/
+
+noncomputable theory
+
+section complex_conformal_linear_map
+
+variables {f : ℂ → ℂ} {z : ℂ} {g : ℂ →L[ℝ] ℂ}
```

Comment:
Can you please make the results apply to `f : ℂ → E` with `E` an arbitrary complex normed space?

Optionally (this would require more energy), to `f : F → E` with `F` an arbitrary 1d complex normed space -- I think this further generalization might come in handy in the manifolds library at some point, if one ever works with `F = (fin 1) → ℂ`.

Reply:
I am not sure how the generalization would work. In that case, we would be needing a general complex conjugate function in order to define antiholomorphic functions.

Reply:
The easiest way is to do only the first generalization I mentioned, and to state the result in the form: either `f` or `f ∘ conj` is holomorphic.

Reply:
My proof of nonzero complex linear maps are conformal linear maps relies on `analysis/complex/isometry`, which is not generalized to arbitrary complex normed space. Is it true that R-linear isometries of the complex plane to some complex normed space are necessarily complex linear?

Reply:
I guess the generalized version is just for one direction.  A (nonzero) complex-linear or complex-antilinear map of C into a complex normed space is conformal, but a conformal map of C into a complex normed space need not necessarily be complex-linear or complex-antilinear.

Reply:
Then should I state the generalized lemma for one direction and prove the opposite direction for linear maps between the complex plane?

Reply:
Yes, that sounds best to me!

Reply:
I have made several generalizations and changed the statement for antiholomorphic functions. I hope they are the ones you have in mind.